pkg/metrics/metrics.go

```diff
@@ -0,0 +1,66 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("registration_metrics")
+var Reg *prometheus.Registry
+
+const (
+	MetricLabelRouted   = "Routed"
+	MetricLabelRejected = "Rejected"
+	MetricLabelListed   = "Listed"
+)
+
+// histogram with labels
+var (
+	// RegServProxyRouteHistogramVec measures the time taken by proxy before forwarding the request
+	RegServProxyRouteHistogramVec *prometheus.HistogramVec
+	// RegServProxyResponseHistogramVec measures the response time for either response or error from proxy when there is no routing
+	RegServProxyResponseHistogramVec *prometheus.HistogramVec
+)
+
+// collections
+var (
+	allHistogramVecs = []*prometheus.HistogramVec{}
+)
+
+const metricsPrefix = "sandbox_"
+
+func init() {
+	initMetrics()
+}
+func initMetrics() {
+	log.Info("initializing custom metrics")
+	RegServProxyRouteHistogramVec = newHistogramVec("proxy_route_time", "time taken by proxy to route ", "routeTo")
+	RegServProxyResponseHistogramVec = newHistogramVec("proxy_response_time", "time for response of a request to proxy ", "responseFor")
```

Comment:
let's have a `status` label (202, 5xx, etc.) and a `verb` label with values `get` or `list` depending on the type of request ("get workspace" or "list workspace")

Reply:
we resolved naming in sync call, and here are the new names [12580c1](https://github.com/codeready-toolchain/registration-service/pull/342/commits/12580c1cb0d601c673b1516f5938ea5ee078414c)

Reply:
my apologies, I forgot about that :/